File: python/main/dp/MinCostClimbingStairs.py
from typing import List


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp1, dp2, dp = 0, 0, 0
        for sc in cost:
            dp = min(dp1 + sc, dp2 + sc)
            dp2 = dp1
            dp1 = dp
        return min(dp2, dp1)
        pass


if __name__ == "__main__":
    solution = Solution()
    output = solution.minCostClimbingStairs([10, 15, 20])
    print(output)

File: java/src/main/java/com/github/algo/dp/MinCostClimbingStairs.java
package com.github.algo.dp;

public class MinCostClimbingStairs {

    public int minCostClimbingStairs(int[] cost) {
        int a = 0, b = 0;
        for (int i = cost.length - 1; i >= 0; i--) {
            int currentCost = cost[i] + Math.min(a, b);
            b = a;
            a = currentCost;
        }
        return  Math.min(a, b);
    }

}
